Improved the BlockChangeFlags.

Cybermaxke opened the pull request:
**API** | [Common](https://github.com/SpongePowered/SpongeCommon/pull/1648)

@gabizou

Cybermaxke committed:
Improved the BlockChangeFlags.

parlough commented:
Can we also add some Javadocs to the `BlockChangeFlags` fields? While the names are fairly indicative of what they do, some may benefit from better explanations.

Cybermaxke committed:
Added docs to the cataloged block change flags.

Cybermaxke commented:
@gabizou I updated this PR for the changes you made, this now just removes the `orFlag` and `exclusiveOrFlag` methods and added some docs to the cataloged block change flags.
